src/PublicApiGenerator/CodeTypeDeclarationExtensions.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PublicApiGenerator
+{
+    internal static class CodeTypeDeclarationExtensions
+    {
+        public static CodeTypeDeclaration Sort(this CodeTypeDeclaration original)
+        {
+            if (original.IsEnum)
+            {
+                return original;
+            }
+
+            var sorted = new CodeTypeDeclaration(original.Name)
+            {
+                Attributes = original.Attributes,
+                CustomAttributes = original.CustomAttributes,
+                IsClass = original.IsClass,
+                IsEnum = original.IsEnum,
+                IsInterface = original.IsInterface,
+                IsPartial = original.IsPartial,
+                IsStruct = original.IsStruct,
+                LinePragma = original.LinePragma,
+                Name = original.Name,
+                TypeAttributes = original.TypeAttributes,
+            };
+
+            sorted.BaseTypes.AddRange(original.BaseTypes);
+            sorted.Comments.AddRange(original.Comments);
+            sorted.EndDirectives.AddRange(original.EndDirectives);
+            sorted.StartDirectives.AddRange(original.StartDirectives);
+            sorted.TypeParameters.AddRange(original.TypeParameters);
+
+            foreach (var key in original.UserData.Keys)
+            {
+                sorted.UserData[key] = original.UserData[key];
+            }
+
+            var emptyParamList = new List<CodeParameterDeclarationExpression>();
+
+            var sortedMembers = original.Members.OfType<CodeTypeMember>()
+                .OrderBy(m => m.GetType().Name, StringComparer.Ordinal)
```

Comment:
Properties go ahead of methods?

Reply:
Yes, this appears to have no effect on the order in comparison to what is currently in `master`.